src/components/screens/SwapScreen/helpers/swapCalculations.ts

```diff
@@ -0,0 +1,108 @@
+import { BigNumber } from "bignumber.js";
+import { DEFAULT_DECIMALS } from "config/constants";
+import { PricedBalance } from "config/types";
+import { formatAssetAmount } from "helpers/formatAmount";
+
+/**
+ * Calculates conversion rate between two amounts
+ * Following the extension's conversion rules
+ */
+export const calculateConversionRate = (
+  fromAmount: string,
+  toAmount: string,
+  conversionRate?: string,
+): string => {
+  if (conversionRate) return conversionRate;
+
+  const fromAmountBN = new BigNumber(fromAmount);
+  const toAmountBN = new BigNumber(toAmount);
+
+  if (fromAmountBN.isZero()) return "0";
+
+  const rate = toAmountBN.dividedBy(fromAmountBN);
+
+  return formatAssetAmount(rate.toFixed(DEFAULT_DECIMALS));
+};
+
+/**
+ * Formats conversion rate for display with proper symbols
+ * Uses formatAssetAmount for consistent 7-decimal formatting following extension rules
+ */
+export const formatConversionRate = (
+  rate: string,
+  fromSymbol: string,
+  toSymbol: string,
+): string => {
+  if (!rate || rate === "0") return "";
+
+  const roundedRate = BigNumber(rate).toFixed(DEFAULT_DECIMALS);
+  const formattedRate = formatAssetAmount(roundedRate);
+
+  return `1 ${fromSymbol} ≈ ${formattedRate} ${toSymbol}`;
+};
+
+/**
+ * Calculates minimum received amount based on slippage
+ */
+export const calculateMinimumReceived = (
+  toAmount: string,
+  allowedSlippage: string,
+  minimumReceived?: string,
+): string => {
+  if (minimumReceived) return minimumReceived;
+
+  const toAmountBN = new BigNumber(toAmount);
+  const slippageMultiplier = BigNumber(1).minus(
+    BigNumber(allowedSlippage).dividedBy(100),
+  );
+
+  return toAmountBN.multipliedBy(slippageMultiplier).toFixed(7);
+};
+
+/**
+ * Formats transaction date for display
+ */
+export const formatTransactionDate = (createdAt?: string): string => {
```

Comment:
I think we are also formatting transaction date in other places like the send flow. If so, we should probably export this to our main helper folder so we could use it anywhere.